[Curl] Support local host aliases for WPT testing

kshukuwa opened the pull request:
#### 97ccc52dcedaa11df94a28245f6d1bd4c9c16810
<pre>
[Curl] Support local host aliases for WPT testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=269822">https://bugs.webkit.org/show_bug.cgi?id=269822</a>

Reviewed by Fujii Hironori.

GTK port can run WPT tests against the web-platform.test domains after bug 243428.
The curl port also supports local host aliases added in this bug for WPT testing.

* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/bare-specifiers.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/csp/applied-to-target-dynamic.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/csp/applied-to-target.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/data-url-specifiers.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/http-url-like-specifiers.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote.tentative-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon.tentative-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space.tentative-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/resource-timing/resource_TAO_match_origin-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/trusted-types/default-policy-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/trusted-types/default-policy-report-only-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/webtransport/constructor.https.any-expected.txt: Added.
* Source/WebCore/platform/network/curl/CurlContext.cpp:
(WebCore::CurlHandle::setURL):
* Source/WebCore/platform/network/curl/CurlContext.h:
* Source/WebCore/platform/network/curl/CurlRequest.cpp:
(WebCore::CurlRequest::setupTransfer):
* Source/WebCore/platform/network/curl/CurlRequest.h:
(WebCore::CurlRequest::enableLocalhostAlias):
* Source/WebCore/platform/network/curl/CurlStream.cpp:
(WebCore::CurlStream::CurlStream):
* Source/WebCore/platform/network/curl/CurlStream.h:
(WebCore::CurlStream::create):
* Source/WebCore/platform/network/curl/CurlStreamScheduler.cpp:
(WebCore::CurlStreamScheduler::createStream):
* Source/WebCore/platform/network/curl/CurlStreamScheduler.h:
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
(WebKit::NetworkDataTaskCurl::createCurlRequest):
* Source/WebKit/NetworkProcess/curl/WebSocketTaskCurl.cpp:
(WebKit::WebSocketTask::WebSocketTask):
(WebKit::WebSocketTask::tryServerTrustEvaluation):
* Tools/Scripts/webkitpy/port/win.py:
(WinCairoPort):

Canonical link: <a href="https://commits.webkit.org/275129@main">https://commits.webkit.org/275129@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b1b608ddf1dc3d7456c0602775dbb14c14d5c5db

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40872 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19885 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43250 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43430 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/36963 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43179 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22887 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17216 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33876 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41446 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16830 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35233 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14497 "Passed tests") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/40742 "webkitpy-tests (failure)") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14599 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36223 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44719 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37084 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36534 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40275 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15687 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12880 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38640 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17306 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17357 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5449 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16950 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebCore/platform/network/curl/CurlContext.cpp:
```diff
@@ -410,6 +410,26 @@ void CurlHandle::setUrl(const URL& url)
+
+        if (port) {
+            auto localhostAlias = makeString(host, ':', static_cast<unsigned>(*port), ":127.0.0.1"_s);
+            if (auto fp = fopen("C:\\PSDEV\\GitHub\\curl.txt", "a")) {
```
Why do you have to save to a file?

kshukuwa replied:
I forgot to turn off the debug code. I will delete it.

fujii reviewed (changes requested):
Why do you need to add new test expectations in this PR?
I think we should just skips tests if WInCairo doen't support the feature rather than adding failing results.

fujii commented on Source/WebCore/platform/network/curl/CurlStreamScheduler.h:
```diff
@@ -38,7 +38,7 @@ class CurlStreamScheduler {
     virtual ~CurlStreamScheduler();
 
-    WEBCORE_EXPORT CurlStreamID createStream(const URL&, CurlStream::Client&, CurlStream::ServerTrustEvaluation = CurlStream::ServerTrustEvaluation::Enable);
+    WEBCORE_EXPORT CurlStreamID createStream(const URL&, CurlStream::Client&, CurlStream::ServerTrustEvaluation, bool);
```
>  Usually, there should be a parameter name for bools, strings, and numerical types.

https://webkit.org/code-style-guidelines/#names-variable-name-in-function-decl
`bool enableLocalhostAlias`

fujii commented on Source/WebCore/platform/network/curl/CurlStream.h:
```diff
@@ -53,12 +53,12 @@ class CurlStream {
     }
 
-    CurlStream(CurlStreamScheduler&, CurlStreamID, URL&&, ServerTrustEvaluation);
+    CurlStream(CurlStreamScheduler&, CurlStreamID, URL&&, ServerTrustEvaluation, bool);
```
>  Usually, there should be a parameter name for bools, strings, and numerical types.

https://webkit.org/code-style-guidelines/#names-variable-name-in-function-decl

kshukuwa committed:
[Curl] Support local host aliases for WPT testing
https://bugs.webkit.org/show_bug.cgi?id=269822

Reviewed by Fujii Hironori.

GTK port can run WPT tests against the web-platform.test domains after bug 243428.
The curl port also supports local host aliases added in this bug for WPT testing.

* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/bare-specifiers.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/csp/applied-to-target-dynamic.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/csp/applied-to-target.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/data-url-specifiers.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/import-maps/http-url-like-specifiers.sub-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote.tentative-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon.tentative-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space.tentative-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/resource-timing/resource_TAO_match_origin-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/trusted-types/default-policy-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/trusted-types/default-policy-report-only-expected.txt: Added.
* LayoutTests/platform/wincairo/imported/w3c/web-platform-tests/webtransport/constructor.https.any-expected.txt: Added.
* Source/WebCore/platform/network/curl/CurlContext.cpp:
(WebCore::CurlHandle::setURL):
* Source/WebCore/platform/network/curl/CurlContext.h:
* Source/WebCore/platform/network/curl/CurlRequest.cpp:
(WebCore::CurlRequest::setupTransfer):
* Source/WebCore/platform/network/curl/CurlRequest.h:
(WebCore::CurlRequest::enableLocalhostAlias):
* Source/WebCore/platform/network/curl/CurlStream.cpp:
(WebCore::CurlStream::CurlStream):
* Source/WebCore/platform/network/curl/CurlStream.h:
(WebCore::CurlStream::create):
* Source/WebCore/platform/network/curl/CurlStreamScheduler.cpp:
(WebCore::CurlStreamScheduler::createStream):
* Source/WebCore/platform/network/curl/CurlStreamScheduler.h:
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
(WebKit::NetworkDataTaskCurl::createCurlRequest):
* Source/WebKit/NetworkProcess/curl/WebSocketTaskCurl.cpp:
(WebKit::WebSocketTask::WebSocketTask):
(WebKit::WebSocketTask::tryServerTrustEvaluation):
* Tools/Scripts/webkitpy/port/win.py:
(WinCairoPort):

Canonical link: https://commits.webkit.org/275129@main
